src/Compilers/Core/Rebuild/CSharpCompilationFactory.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using CS = Microsoft.CodeAnalysis.CSharp;
+
+namespace BuildValidator
+{
+    public sealed class CSharpCompilationFactory : CompilationFactory
+    {
+        public override ParseOptions ParseOptions => CSharpParseOptions;
+        public override CompilationOptions CompilationOptions => CSharpCompilationOptions;
+        public CSharpParseOptions CSharpParseOptions { get; }
+        public CSharpCompilationOptions CSharpCompilationOptions { get; }
+
+        private CSharpCompilationFactory(
+            string assemblyFileName,
+            CompilationOptionsReader optionsReader,
+            CSharpParseOptions parseOptions,
+            CSharpCompilationOptions compilationOptions)
+            : base(assemblyFileName, optionsReader)
+        {
+            Debug.Assert(optionsReader.GetLanguageName() == LanguageNames.CSharp);
+            CSharpParseOptions = parseOptions;
+            CSharpCompilationOptions = compilationOptions;
+        }
+
+        public static new CSharpCompilationFactory Create(string assemblyFileName, CompilationOptionsReader optionsReader)
```

Comment:
>CSharpCompilationFactory Create [](start = 26, length = 31)

Should this method be used by the base class `Create()` method only? If so, consider making this `internal`.

Reply:
good point